static/app/api/apiOptions.spec.ts

```diff
@@ -0,0 +1,164 @@
+import {expectTypeOf} from 'expect-type';
+
+import type {ApiResult} from 'sentry/api';
+import type {Release} from 'sentry/types/release';
+
+import {apiOptions} from './apiOptions';
+
+type QueryFunctionResult<T> = Promise<ApiResult<T>> | ApiResult<T>;
+
+describe('apiOptions', () => {
+  test('should encode path parameters correctly', () => {
+    const options = apiOptions(
+      '/projects/$orgSlug/$projectSlug/releases/$releaseVersion/',
+      {
+        staleTime: 0,
+        path: {
+          orgSlug: 'my-org',
+          projectSlug: 'my-project',
+          releaseVersion: 'v 1.0.0',
+        },
```

Comment:
every `$` parameter (we can choose a different identifier too) needs to be passed in, and the value needs to be `string | number`. Excess properties are not allowed. The full url will be built for you under the hood, and values will be properly url encoded.

This extends the current approach of passing query params as `query`, but those are not yet type-safe (I’ve just re-used the previous type).